### PR TITLE
Initial Travis CI config

### LINKED ISSUE #6: Set up Travis CI
It should run `gulp test` and `gulp lint` for incoming PRs.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @gauntface @wibblymat @addyosmani 

Based on the Travis CI setup that's [being used for `sw-toolbox`](https://github.com/GoogleChrome/sw-toolbox/blob/master/.travis.yml).

Fixes #6

### CODE REVIEW

**Comment by jeffposnick** [comment]
There are Firefox Beta failures due to https://github.com/GoogleChrome/sw-testing-helpers/issues/36

I'm going to omit Firefox beta from the Travis CI environment until @gauntface has had a chance to come up with a recommended way of testing it.

**Comment by gauntface** [comment]
I think I know how to fix ff beta, expect a fix early next week after I/O, will do a PR for either updating sw-testing helpers and / or the travis yaml.

LGTM for this PR

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  # Get firefbox beta and extract to tar for the test runner to use.
+  # - wget 'https://download.mozilla.org/?product=firefox-beta-latest&lang=en-US&os=linux64' -O firefox-beta.tar.bz2 && tar xvjf firefox-beta.tar.bz2
+
+# Read more here: https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-a-GUI
+before_script:
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+
+script:
+  - gulp build && gulp lint test
```
Comment (gauntface) [inline .travis.yml]: But: could this be gulp build lint test ?

Comment (jeffposnick) [inline .travis.yml]: I want the `build` to take place first, and if I list them all together then they'll be run concurrently. If there's a better way to get the `build` to run first (maybe via one of the other lifecycle events in Travis?) I'm happy to switch that out, but this seemed to work.

Comment (jeffposnick) [inline .travis.yml]: Merging for now. If there are suggestions for improving the automation of the `build` step, let me know via a new issue or PR.

**Pull request merged**